### PR TITLE
fix: no longer unpacking and flattening payload, closes #7673

### LINKED ISSUE #7673: [bug] app unresponsive after some plugin API calls (tauri 2.0)
### Describe the bug

I have a tauri-2.0 project project on Windows 11 that uses (amon others) the "dialog" and "window" plugins. With version 2.0.0-alpha.0 of those plugins and tauri 2.0.0-alpha.10, everything worked as expected.

After updating to

```
[build-dependencies]
tauri-build = { version = "2.0.0-alpha.8", features = [] }

[dependencies]
serde_json = "1.0"
serde = { version = "1.0", features = ["derive"] }
tauri = { version = "2.0.0-alpha.11", features = [] }
tauri-plugin-window = "2.0.0-alpha.1"
tauri-plugin-dialog = "2.0.0-alpha.1"
```

the app (browser **and** dev-tools) becomes completely unresponsive after some API-calls.

### Reproduction

* Check out [dtanzer/tauri-reproduction-example](https://github.com/dtanzer/tauri-reproduction-example) / commit (08ac11c57f7826cb038f07f5f72ede21cbcff242)[https://github.com/dtanzer/tauri-reproduction-example/commit/08ac11c57f7826cb038f07f5f72ede21cbcff242]
* Install the dependencies and start the App
* Click the buttons "Open Directory" or "Is Maximized?" -> The app becomes unresponsive and does not recover (The third button works for me)

### Expected behavior

The app continues to accept user input after an API call

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.22621 X64
    ✔ WebView2: 115.0.1901.203
    ✔ MSVC:
        - Visual Studio Build Tools 2017
        - Visual Studio Build Tools 2022
    ✔ rustc: 1.71.0 (8ede3aae2 2023-07-12)
    ✔ Cargo: 1.71.0 (cfd3bbd8f 2023-06-08)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)        
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.5.0
    - npm: 9.8.0

[-] Packages
    - tauri [RUST]: 2.0.0-alpha.11
    - tauri-build [RUST]: 2.0.0-alpha.8
    - wry [RUST]: 0.31.1
    - tao [RUST]: 0.22.2
    - @tauri-apps/api [NPM]: 2.0.0-alpha.6
    - @tauri-apps/cli [NPM]: 2.0.0-alpha.11        

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by simonhyll**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information
Issue: #7673

### CODE REVIEW

**Review by JonasKruckenberg — DISMISSED** [review]
Even if this doesn't fix the bug in question it's a good change 👍

File: .changes/fix-ipc-payload.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"tauri": 'patch:bug'
+---
+
+No longer unpacking and flattening the `payload` over the IPC so that commands with an argument called `options` don't break.
```
Comment (JonasKruckenberg) [inline .changes/fix-ipc-payload.md]: It's actually also parameters named `error`, `cmd`, or `callback`

Comment (simonhyll) [reply .changes/fix-ipc-payload.md]: Updated

**Review by JonasKruckenberg — APPROVED** [review]

**Comment by simonhyll** [comment]
A user reported this only fixes it on Linux, so I'll see if I can figure out the Windows issue as well. Can't do much about Mac.

**Comment by nikoPLP** [comment]
> A user reported this only fixes it on Linux, so I'll see if I can figure out the Windows issue as well. Can't do much about Mac.

Tested right now on macos, it fixes the use of `options` in the command args.

**Comment by simonhyll** [comment]
I can't for the life of me figure out why Windows is crashing. I'd say merge this and someone else take over making Windows work.

**Comment by reyamir** [comment]
sadly, this issue really serious

**Comment by amrbashir** [comment]
Flattening is indeed bad, I think there was an issue with it in v1, and so we shouldn't flatten at all.
I would still wait for @lucasfernog to merge this since he is the one familiar with this piece of code.

**Review by lucasfernog — APPROVED** [review]

**Comment by LetrixZ** [comment]
I tried this on macOS with the `dialog` plugin and it still freezes. Seems to be an issue with the custom protocol that only runs on Windows and macOS.
Linux works as expected.

**Pull request merged**